src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -253,231 +261,110 @@ private void CheckAnalysisWasRequested()
-                .Returns(optionalDataToReturn ?? captured);
-            capturedFilterInput = captured;
-        }
+            issueConsumer.Invocations.Clear();
+            issueConsumerFactory.Invocations.Clear();
 
-        private void CheckErrorListRefreshWasRequestedOnce(IssuesSnapshotFactory factory)
-        {
-            mockSonarErrorDataSource.Verify(x => x.RefreshErrorList(factory), Times.Once);
-        }
+            testSubject.RequestAnalysis(Mock.Of<IAnalyzerOptions>());
 
-        private static IAnalysisIssueVisualization CreateIssue(string ruleKey, int startLine, int endLine)
-        {
-            var issue = new DummyAnalysisIssue
-            {
-                RuleKey = ruleKey,
-                PrimaryLocation = new DummyAnalysisIssueLocation
-                {
-                    TextRange = new DummyTextRange
-                    {
-                        StartLine = startLine,
-                        EndLine = endLine,
-                    }
-                }
-            };
-
-            var issueVizMock = new Mock<IAnalysisIssueVisualization>();
-            issueVizMock.Setup(x => x.Issue).Returns(issue);
-            issueVizMock.Setup(x => x.Location).Returns(issue.PrimaryLocation);
-            issueVizMock.Setup(x => x.Flows).Returns(Array.Empty<IAnalysisIssueFlowVisualization>());
-            issueVizMock.SetupProperty(x => x.Span);
-            issueVizMock.Object.Span = new SnapshotSpan(CreateMockTextSnapshot(1000, "any line text").Object, 0, 1);
-
-            return issueVizMock.Object;
-        }
+            // The factory should have been called...
+            issueConsumerFactory.Verify(x => x.Create(mockedJavascriptDocumentFooJs.Object, It.IsAny<string>(), It.IsAny<Guid>(), It.IsAny<PublishSnapshot>()), Times.Once);
 
-        private static IAnalysisIssueVisualization CreateFileLevelIssue(string ruleKey)
-        {
-            var issue = new DummyAnalysisIssue
-            {
-                RuleKey = ruleKey,
-                PrimaryLocation = new DummyAnalysisIssueLocation { TextRange = null }
-            };
-
-            var issueVizMock = new Mock<IAnalysisIssueVisualization>();
-            issueVizMock.Setup(x => x.Issue).Returns(issue);
-            issueVizMock.Setup(x => x.Location).Returns(issue.PrimaryLocation);
-            issueVizMock.Setup(x => x.Flows).Returns(Array.Empty<IAnalysisIssueFlowVisualization>());
-
-            return issueVizMock.Object;
+            // and the return issue consumer should have been cleared
+            issueConsumer.Verify(x => x.Accept("foo.js", Enumerable.Empty<IAnalysisIssue>()), Times.Once);
         }
 
-
-        #endregion
-
-        #region File without associated project tests
-
         [TestMethod]
-        public void WhenSubjectIsCreated_FileHasNoAssociatedProject_NoExceptionIsThrown()
+        public void RequestAnalysis_FileHasNoAssociatedProject_NoExceptionIsThrown()
```

Comment:
Existing test - fixed up to check that it passes the expected project name and guid values to the `IssueConsumerFactory` if the file isn't part of a project.